x-pack/plugins/maps/public/shared/layers/sources/es_search_source/es_search_source.js

```diff
@@ -127,7 +135,64 @@ export class ESSearchSource extends AbstractESSource {
     ];
   }
 
-  async getGeoJsonWithMeta(layerName, searchFilters) {
+  async _getTopHits(layerName, searchFilters) {
+    const {
+      topHitsSplitField,
+      topHitsTimeField,
+      topHitsSize,
+    } = this._descriptor;
+
+    const searchSource = await this._makeSearchSource(searchFilters, 0);
+    searchSource.setField('aggs', {
+      entitySplit: {
+        terms: {
+          field: topHitsSplitField,
+          size: 10000
+        },
+        aggs: {
+          entityHits: {
+            top_hits: {
+              sort: [
+                {
+                  [topHitsTimeField]: {
+                    order: 'desc'
+                  }
+                }
+              ],
+              _source: {
+                includes: searchFilters.fieldNames
+              },
+              size: topHitsSize
+            }
+          }
+        }
+      }
+    });
+
+    const resp = await this._runEsQuery(layerName, searchSource, 'Elasticsearch document top hits request');
+
+    let hasTrimmedResults = false;
+    const allHits = [];
+    const entityBuckets = _.get(resp, 'aggregations.entitySplit.buckets', []);
+    entityBuckets.forEach(entityBucket => {
+      const { total, hits } = _.get(entityBucket, 'entityHits.hits', {});
+      // Reverse hits list so they are drawn from oldest to newest (per entity) so newest events are on top
+      allHits.push(...hits.reverse());
+      if (total > hits.length) {
```

Comment:
`allHits` and `total` could be null here which would throw errors for lines 180-181. Should we assign them default values?